docs/en/observability/profiling-add-symbols.asciidoc

```diff
@@ -0,0 +1,151 @@
+
+If you don't mind deploying your applications with debug symbols, run:
+
+[source,bash]
+----
+source config.txt
+./elastic-profiling push-symbols executable -e ./my-c-app 
+----
+
+If you don't want debug symbols in production, copy the executable and strip the copy.
+You can then use the `-d` argument to instruct the tool to read the symbols from the original
+unstripped binary while still calculating the file hash from the final stripped binary. After
+the symbols have been pushed, you can remove the unstripped binary:
+
+[source,bash]
+----
+cp ./my-app ./my-stripped-app
+strip ./my-stripped-app
+source config.txt
+./elastic-profiling push-symbols executable -e ./my-stripped-app -d ./my-app
+rm ./my-app
+----
+
+WARNING: Pushing debug information and then stripping the binary later **does not work**.
+The executable passed using the `-e` argument is used to calculate the file hash that associates stack traces with their symbols and stripping the binary later changes that hash.
+
+[discrete]
+[[profiling-symbols-linux]]
+== Linux distribution packages
+
+For Debian, Ubuntu, Fedora, and Arch Linux, the `elastic-profiling` tool supports pushing symbols for an entire package at once. 
+The required debug symbols are automatically retrieved from the distribution's https://wiki.debian.org/Debuginfod[debuginfod] server.
+
+For this to work, make sure that the `debuginfod`` client is installed on your machine.
+
+[horizontal]
+*Distribution*:: *Install Command*
+Debian:: `sudo apt install debuginfod`
+Ubuntu:: `sudo apt install debuginfod`
+Fedora:: `sudo dnf install elfutils-debuginfod-client`
+Arch:: `sudo pacman -S debuginfod` 
+
+
+Then, invoke `elastic-profiling` as follows:
+
+[source,bash]
+----
+source config.txt
+./elastic-profiling push-symbols package -p package-name
+----
+
+For example, to push symbols for libc on Debian:
+
+[source,bash]
+----
+source config.txt
+./elastic-profiling push-symbols package -p libc6
+----
+
+NOTE: debuginfod servers for many distributions can be unreliable and may result in the tool printing `No debug info found for executable, skipping.` for all executables. This is more common if you're on a faster moving release-channel of your distribution (for example, Debian Testing).
+
+[discrete]
+[[profiles-limitations]]
+== Native symbolization limitations
+
+In the current beta release of Elastic Universal Profiling, native symbolization is still limited
+in a few important ways:
+
+* No virtual frames for inline functions
+* No symbols for leaf frames (top-most frame in a trace)
+* No automatic insertion of debug symbols for OS packages
+
+We hope to improve these limitations and remove them in later versions.
+
+NOTE: If symbols are not displayed in Kibana after ingesting them, try restarting Kibana in your deployment (Cloud → Deployments → `<Deployment Name>` → Kibana → Force Restart). This is a known issue in 8.5 and will be fixed in later versions. 
```

Comment:
This note is about a bug in 8.5.0 and I believe it can be removed. @jbcrail can probably advise whether this is definitely fixed.

Reply:
Unfortunately it is still theoretically possible to encounter the bug, but it's less likely now that the Elasticsearch plugin is enabled by default in 8.7 (there are no caches in the path that uses the Elasticsearch plugin). If the plugin was disabled and `elastic-profiling push-symbols` failed to clear the caches, then symbols may not be displayed within the 5-minute TTL window on those caches.

I added a note in https://github.com/elastic/prodfiler/issues/2994 to remove the reference to the bug in this document after the caches are removed.